dcp-0012/dcp-0012.mediawiki

```diff
@@ -13,5 +13,436 @@ Replaces: DCP0010
+	subsidy := int64(mainnetBaseSubsidy)
+	for i := int64(0); i < reductions; i++ {
+		subsidy *= mainnetSubsidyReductionMultiplier
+		subsidy /= mainnetSubsidyReductionDivisor
+
+		// Stop once no further reduction is possible.  This ensures a bounded
+		// computation for large requested intervals and that all future
+		// requests for intervals at or after the final reduction interval
+		// return 0 without recalculating.
+		if subsidy == 0 {
+			break
+		}
+	}
+	return subsidy
+}
+</pre>
+
+===Work (PoW) Proportion Subsidy Calculation===
+
+<pre>
+// calcWorkSubsidyDCP0012 returns the proof of work subsidy for a block at the
+// provided height and given number of votes using the round 2 modified subsidy
+// split.
+func calcWorkSubsidyDCP0012(height int64, voters uint16) int64 {
+	// These parameters are ordinarily unique per chain and thus should be
+	// passed into the function via a chain parameters structure, however,
+	// they are defined as constants with the mainnet parameters here for the
+	// purposes of providing a self-contained function for the DCP.
+	const (
+		mainnetStakeValidationHeight = 4096
+		votesPerBlock                = 5
+		minVotesRequired             = (votesPerBlock / 2) + 1
+	)
+
+	// The first block has special subsidy rules.
+	if height == 1 {
+		return calcBlockSubsidy(height)
+	}
+
+	// The subsidy is zero if there are not enough voters once voting begins.  A
+	// block without enough voters will fail to validate anyway.
+	if height >= mainnetStakeValidationHeight && voters < minVotesRequired {
+		return 0
+	}
+
+	// Calculate the full block subsidy and reduce it according to the PoW
+	// proportion.
+	const proportion = 1
+	const totalProportions = 100
+	subsidy := calcBlockSubsidy(height)
+	subsidy *= proportion
+	subsidy /= totalProportions
+
+	// Ignore any potential subsidy reductions due to the number of votes prior
+	// to the point voting begins.
+	if height < mainnetStakeValidationHeight {
+		return subsidy
+	}
+
+	// Adjust for the number of voters.
+	return (int64(voters) * subsidy) / votesPerBlock
+}
+</pre>
+
+===Stake Vote (PoS) Proportion Subsidy Calculation===
+
+<pre>
+// calcStakeVoteSubsidyDCP0012 returns the subsidy for a single stake vote for a
+// block at the provided height using the round 2 modified subsidy split.
+func calcStakeVoteSubsidyDCP0012(height int64) int64 {
+	// These parameters are ordinarily unique per chain and thus should be
+	// passed into the function via a chain parameters structure, however,
+	// they are defined as constants with the mainnet parameters here for the
+	// purposes of providing a self-contained function for the DCP.
+	const (
+		mainnetStakeValidationHeight = 4096
+		votesPerBlock                = 5
+		minVotesRequired             = (votesPerBlock / 2) + 1
+	)
+
+	// Votes have no subsidy prior to the point voting begins.  The minus one
+	// accounts for the fact that vote subsidy are, based on the height that is
```

Comment:
```suggestion
	// accounts for the fact that vote subsidies are, based on the height that is
```

Reply:
Corrected and removed the unnecessary comma.